### PR TITLE
Add methods of squid v0.3 spec

### PR DESCRIPTION
**Pull request by eruizgar91**
## Description
- Get list of the Assets Published by an user
-  Get list of the Assets Purchased by an user
- Get the Owner of an Asset
- Expose validate metadata of Plecos and allow retire assets of Aquarius.

## Is this PR related with an open issue?

Related to Issue #102

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] Follows the code style of this project.
- [x] Tests Cover Changes
- [x] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

File: src/main/java/com/oceanprotocol/squid/models/asset/AssetMetadata.java [hunk 1]
```diff
@@ -94,10 +94,10 @@ public static AssetMetadata builder()   {
         public String inLanguage;
 
         @JsonProperty
-        public String tags;
+        public ArrayList<String> tags;
 
         @JsonProperty
-        public String price;
+        public Integer price;
```
Comment (aaitor) [inline src/main/java/com/oceanprotocol/squid/models/asset/AssetMetadata.java]: if it's not a String probably the price should be a BigInteger, an integer is too small and couldn't represent prices

File: src/main/java/com/oceanprotocol/squid/external/AquariusService.java [hunk 2]
```diff
@@ -200,4 +201,17 @@ public SearchResult searchDDO(SearchQuery searchQuery) throws DDOException {
 
     }
 
+    public boolean retireAssetDDO(String id) throws Exception {
```
Comment (aaitor) [inline src/main/java/com/oceanprotocol/squid/external/AquariusService.java]: This function needs documentation

File: src/main/java/com/oceanprotocol/squid/external/AquariusService.java [hunk 3]
```diff
@@ -200,4 +201,17 @@ public SearchResult searchDDO(SearchQuery searchQuery) throws DDOException {
...
+        HttpResponse response= HttpHelper.httpClientDelete(this.ddoEndpoint + "/" + id);
+        if (response.getStatusCode() == 200 || response.getStatusCode() == 201)    {
+            return true;
+        }
+        throw new Exception("Unable to update DDO: " + response.toString());
+    }
+
+    public boolean validateMetadata(AssetMetadata metadata) throws Exception{
```
Comment (aaitor) [inline src/main/java/com/oceanprotocol/squid/external/AquariusService.java]: Same for this one

File: src/main/java/com/oceanprotocol/squid/manager/AssetsManager.java [hunk 4]
```diff
@@ -108,5 +109,14 @@ public SearchResult searchAssets(Map<String, Object> params, int offset, int pag
         return getAquariusService().searchDDO(searchQuery);
     }
 
+    public Boolean deleteAsset(DID did) throws Exception {
```
Comment (aaitor) [inline src/main/java/com/oceanprotocol/squid/manager/AssetsManager.java]: Documentation

File: src/main/java/com/oceanprotocol/squid/manager/AssetsManager.java [hunk 5]
```diff
@@ -108,5 +109,14 @@ public SearchResult searchAssets(Map<String, Object> params, int offset, int pag
...
+        return getAquariusService().retireAssetDDO(did.getDid());
+
+    }
+
+    public Boolean validateMetadata(AssetMetadata metadata) throws Exception {
```
Comment (aaitor) [inline src/main/java/com/oceanprotocol/squid/manager/AssetsManager.java]: Documentation

File: src/test/java/com/oceanprotocol/squid/api/AssetsApiIT.java [hunk 6]
```diff
@@ -198,4 +201,71 @@ public void query() throws Exception {
...
+
+        oceanAPI.getAssetsAPI().create(metadataBase, providerConfig);
+        log.debug("DDO registered!");
+
+        int assetsOwnedAfter = oceanAPI.getAssetsAPI().ownerAssets(oceanAPI.getMainAccount().address).size();
+        assertEquals(assetsOwnedAfter, assetsOwnedBefore + 1);
+    }
+
+    @Test
+    public void consumerAssets() throws Exception{
```
Comment (aaitor) [inline src/test/java/com/oceanprotocol/squid/api/AssetsApiIT.java]: To save some time in testing, maybe we could join the consume and the consumerAssets test

Comment (eruizgar91) [reply src/test/java/com/oceanprotocol/squid/api/AssetsApiIT.java]: That sounds good to me

**Pull request merged**